src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -217,25 +234,27 @@ func (s *service) Health(ctx thrift.Context) (*rpc.NodeHealthResult_, error) {
 	health := s.health
 	s.RUnlock()
 
-	// Update bootstrapped field if not up to date. Note that we use
-	// IsBootstrappedAndDurable instead of IsBootstrapped to make sure
-	// that in the scenario where a topology change has occurred, none of
-	// our automated tooling will assume a node is healthy until it has
-	// marked all its shards as available and is able to bootstrap all the
-	// shards it owns from its own local disk.
-	bootstrapped := s.db.IsBootstrappedAndDurable()
-
-	if health.Bootstrapped != bootstrapped {
-		newHealth := &rpc.NodeHealthResult_{}
-		*newHealth = *health
-		newHealth.Bootstrapped = bootstrapped
-
-		s.Lock()
-		s.health = newHealth
-		s.Unlock()
-
-		// Update response
-		health = newHealth
+	if s.db != nil {
```

Comment:
This is racey to read without an RLock, because you can't guarantee it's not currently being changed from nil to not nil as you read this (since no RLock held).